api/endpoints/v2/tasks.py

```
@@ -0,0 +1,513 @@
+    stats = task_crud.get_task_stats(db, user_id=user_id)
+    return StandardResponse(
+        message="Task statistics retrieved successfully",
+        data=stats
+    )
+
+
+@router.post("/hierarchy/{parent_id}/{child_id}", response_model=StandardResponse)
+def create_task_hierarchy(
+        parent_id: int,
+        child_id: int,
+        db: Session = Depends(get_db)
```

Ruff (0.15.0)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)